filetracker/servers/storage.py

```diff
@@ -277,10 +340,27 @@ def _exclusive_lock(path):
         os.close(fd)
 
 
+@contextlib.contextmanager
+def _no_lock():
+    """Does nothing, just runs the code within the `with` statement.
+       Used for conditional locking."""
+    yield
+
+
 def _makedirs(path):
     """A py2 wrapper for os.makedirs() that simulates exist_ok=True flag."""
     try:
         os.makedirs(path)
     except OSError as e:
         if e.errno != errno.EEXIST:
             raise
+
+
+def lutime(path, time):
+    if six.PY2:
+        t = email.utils.formatdate(time)
+        if subprocess.call(["touch", "-c", "-h", "-d", t, path]) != 0:
+            raise RuntimeError
```

Comment:
Really there's no way to easily implement touch in Py2?
(BTW. Single quotes for untranslatable string)

Reply:
We both searched for it and the only other solution seems to be calling a syscall manually with ctypes.
Python 2 follows symlinks in utime and doesn't have lutime so there doesn't seem to be any other option. At least we are not aware of any after searching for a while.

Reply:
OK